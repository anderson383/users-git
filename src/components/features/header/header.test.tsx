
import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import Header from '.';
import { BrowserRouter } from 'react-router-dom';

describe('Tests in Header component', () => {
  const setup = () => {
    return render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
  };

  test('Should render the component', async () => {
    const {
      asFragment
    } = setup();

    const buttonImage = screen.getByTestId('img_butt');

    fireEvent.click(buttonImage);
    expect(asFragment()).toMatchSnapshot();
  });
});
