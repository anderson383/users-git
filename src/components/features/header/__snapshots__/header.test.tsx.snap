// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Header component Should render the component 1`] = `
<DocumentFragment>
  <header
    class="header_container"
  >
    <div
      class="header  container"
    >
      <img
        alt=""
        data-testid="img_butt"
        src="/icons/icon-github.svg"
        width="50"
      />
      <nav
        class="menu"
      >
        <h2>
          Test FrontEnd
        </h2>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
